deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.kt

```diff
@@ -38,13 +38,64 @@ data class DeepLinkMatchResult(
             "parameters: $parameterMap"
     }
 
+    /**
+     * Whatever template has the first placeholder (and then configurable path segment) is the less
+     * concrete one.
+     * Because if they would have been all in the same index those elements would have been on the
+     * same level and in the same "list" of elements we compare in order.
+     * In this case the one with the more concete element would have won and the same is true here.
+     */
+    override fun compareTo(other: DeepLinkMatchResult): Int {
```

Comment:
I just moved the compare logic from `DeepLinkMatchResult` to `DeepLinkEntry` as I wanted to use that first but then ended up not doing it, however the change still makes sense as the compare is just based on the `DeepLinkEntry` anyway.